storage_service/locations/api/resources.py

```diff
@@ -839,6 +840,23 @@ def aip_store_callback_request(self, request, bundle, **kwargs):
         else:
             return http.HttpNoContent()
 
+    @_custom_endpoint(expected_methods=['post'])
+    def reindex_request(self, request, bundle, **kwargs):
+        """Index file data from the Package transfer METS file."""
+        package = bundle.obj
+        if package.package_type != Package.TRANSFER:
+            return http.HttpBadRequest(json.dumps({"error": True, "message": "This package is not a transfer."}), content_type='application/json')
+        if package.current_location.purpose != Location.BACKLOG:
+            return http.HttpBadRequest(json.dumps({"error": True, "message": "This package is not part of a transfer backlog location."}), content_type='application/json')
+        try:
+            package.index_file_data_from_transfer_mets()  # Create File entries for every file in the transfer
+        except StorageException as e:
+            LOGGER.warning("Transfer METS data could not be read: %s", str(e))
+            return http.HttpApplicationError(json.dumps({"error": True, "message": "Transfer METS data could not be read."}), content_type='application/json')
```

Comment:
Both.  I normally recommend catching the smallest exception we know of, but since we don't know what exceptions to expect and getting a 500 error in response to an API call sucks, I think changing it to catch `Exception` and the error message be more generic ("An error occurred while reindexing the Transfer" maybe?) would be good.

It's not well documented, but you can pass `exc_info=True` to `LOGGER.warning` (or any of the other logging functions), and it will print the traceback (at least inside an except block), which is super useful.